torch/distributed/fsdp/flat_param.py

```diff
@@ -265,6 +268,7 @@ def _init_metadata(
         self._modules = set(pi.module for pi in self._param_infos).union(
             set(spi.module for spi in self._shared_param_infos)
         )
+        self._root_modules = _get_root_modules(self._modules)
```

Comment:
nit: might be useful to add a comment describing why this property is needed (same goes for other properties as well).

Reply:
Added explanation in the `FlatParameter` docstring